docs/changehistory/NextVersion.md

```diff
@@ -334,6 +347,80 @@ the widget in a floating container. The property `defaultFloatingPosition` may a
 
 The method `getFloatingWidgetContainerIds()` has been added to FrontstageDef to retrieve the Ids for all floating widget containers for the active frontstage as specified by the `frontstageDef`. These ids can be used to query the size of the floating container via `frontstageDef.getFloatingWidgetContainerBounds`. The method `frontstageDef.setFloatingWidgetContainerBounds` can then be used to set the size and position of a floating widget container.
 
+#### `ControlledTree` API Changes
+
+There were some breaking API changes related to `ControlledTree` component:
+
+* The component now takes `TreeModel` rather than `VisibleTreeNodes` as a prop to avoid requiring consumers to manage `VisibleTreeNodes` object. As a result, the `useVisibleTreeNodes` hook was replaced with `useTreeModel` hook. Typical mitigation:
+
+  **Before:**
+
+  ```tsx
+  const visibleNodes = useVisibleTreeNodes(modelSource);
+  return (
+    <ControlledTree {...otherProps} visibleNodes={visibleNodes} />
+  );
+  ```
```

Comment:
The code fragment could be shorter:
```tsx
const visibleNodes = useVisibleTreeNodes(modelSource);
return <ControlledTree visibleNodes={visibleNodes} {...otherProps} />
```

Also, maybe put the relevant prop before the rest.

Reply:
Done, thanks